utils/tracefs.c

```diff
@@ -1,67 +1,59 @@
 #include <fcntl.h>
+#include <linux/magic.h>
+#include <mntent.h>
 #include <stdio.h>
 #include <string.h>
+#include <sys/vfs.h>
 
 #include "utils/tracefs.h"
 #include "utils/utils.h"
 
-#define PROC_MOUNTINFO "/proc/self/mountinfo"
+#define TRACEFS_DIR_PATH "/sys/kernel/tracing"
 
 static char *TRACING_DIR = NULL;
 
 static bool find_tracing_dir(void)
 {
-	FILE *fp;
-	char *line = NULL, fs_type[NAME_MAX], mount_point[PATH_MAX];
-	static char debugfs_suffix[] = "tracing";
-	bool debugfs_found = false;
-	size_t len;
-
-	if (TRACING_DIR)
-		return false;
```

Comment:
You removed this caching logic..

Reply:
I'll revert the caching logic.

Reply:
Hmm.. I'm not sure why the original code returns false.

Reply:
I couldn't figure out why the original code was returning false either. 

The original code works fine, but I think it's semantically correct for the caching logic to return true.